src/dinterpret.d

```diff
@@ -4198,11 +4198,16 @@ public:
 
             if (newval.op == TOKslice)
             {
-                SliceExp se = cast(SliceExp)newval;
-                Expression aggr2 = se.e1;
-                if (aggregate == aggr2)
+                auto se = cast(SliceExp)newval;
+                auto aggr2 = se.e1;
+                const srclower = se.lwr.toInteger();
+                const srcupper = se.upr.toInteger();
+
+                if (aggregate == aggr2 &&
+                    lowerbound < srcupper && srclower < upperbound)
                 {
-                    e.error("overlapping slice assignment [%d..%d] = [%llu..%llu]", lowerbound, upperbound, se.lwr.toInteger(), se.upr.toInteger());
+                    e.error("overlapping slice assignment [%d..%d] = [%llu..%llu]",
```

Comment:
lowerbound is an int, and upperbound is a size_t. Both need different formats. But I'll pull this and fix it myself, since Kenji doesn't seem to be around lately.